.github/workflows/haskell-nix.yml

```diff
@@ -0,0 +1,25 @@
+name: HaskellNix
+
+on: [pull_request]
+jobs:
+  nix:
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        ghc: ['ghc8104']
+        os: [ubuntu-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: true
+    - uses: cachix/install-nix-action@v12
+      with:
+        nix_path: nixpkgs=channel:nixos-20.03
+    - uses: cachix/cachix-action@v8
+      with:
+        name: haskell-language-server
+        authToken: ${{ secrets.HLS_CACHIX_AUTH_TOKEN }}
```

Comment:
The `haskell-language-server` Cachix account is limited to 5GB, so this won't work. I recommend using a different Cachix account for the haskell.nix artifacts, I can set up the token if you share it via email (pepeiborra@gmail.com). 

To test it, you will have to open the PR from a GitHub.com/haskell/haskell-language-server branch, otherwise the GitHub action cannot see the token. I've given you push access

Reply:
I've set up the Cachix account `haskell-nix-language-server` and the secret `HLS_NIX_CACHIX_AUTH_TOKEN`

Reply:
> I can set up the token if you share it via email

Sorry, I just saw this.  Do you still need me to do this?

> I've given you push access

I tried pushing the hkm/haskell-nix branch to haskell/haskell-language-server, but got:

```
ERROR: Permission to haskell/haskell-language-server.git denied to hamishmack.
Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.
```

It used quite a lot of github minutes building GHC.  Is it possible (and would it make sense) to use the [IOHK cache](https://input-output-hk.github.io/haskell.nix/tutorials/getting-started/#setting-up-the-binary-cache) instead of cachix (or better yet as an upstream cache if that is possible)?

Reply:
Github says that it is awaiting your response: 
![image](https://user-images.githubusercontent.com/26626/108787455-b7c5a300-756d-11eb-8f6a-b7ba96ab8347.png)

It makes sense to use the IOHK cache, as long as it freely accessible and also provides Mac OS builds?

Reply:
@hamishmack hi, some update on this?